install.sh

```diff
@@ -25,58 +26,80 @@ pip3 install pexpect
 pip3 install IKP3db
 pip install ikpdb
 
-echo "create folder structure"
+echo "###removing /etc/c9 if exists###"
+WORKING_DIR=/etc/c9
+if [ -d "$WORKING_DIR" ]; then rm -Rf $WORKING_DIR; fi
+
+echo "###create folder structure###"
 mkdir /etc/c9
 mkdir /etc/c9/sock
 mkdir /etc/c9/util
 
-echo "download utils"
+echo "###download utils###"
 wget https://raw.githubusercontent.com/kikito/md5.lua/master/md5.lua -O /etc/c9/util/md5.lua
 wget https://raw.githubusercontent.com/HuJK/Cloud9Hub/master/pip2su.py -O /etc/c9/util/pip2su.py
 wget https://raw.githubusercontent.com/HuJK/Cloud9-hub/master/create_login.py -O /etc/c9/util/create_login.py
 
-echo "install c9 sdk by offical installitaion script"
+echo "###install c9 sdk by offical installitaion script###"
 cd /etc/c9
 HOME=/etc/c9
 git clone https://github.com/c9/core sdk
 cd sdk
 ./scripts/install-sdk.sh
 
-echo "set permission"
+echo "###set permission###"
 chmod -R 755 /etc/c9/sdk
 chmod -R 755 /etc/c9/.c9
 chmod -R 755 /etc/c9/util
 chmod -R 773 /etc/c9/sock
 HOME=/root
+
 # Install npm back
 apt-get install -y npm
 
-echo "add nginx to shadow to make pam_module work"
+echo "###checking username if exists###"
+id -u nginx &>/dev/null || useradd nginx
+id -u www-data &>/dev/null || useradd www-data
+
+echo "###add nginx to shadow to make pam_module work###"
 usermod -aG shadow nginx
 usermod -aG shadow www-data
 
-echo "install Cloud9-hub to nginx config"
+echo "###removing Cloud9-hub to nginx config if exists###"
+rm -f /etc/nginx/sites-available/c9io
+
+echo "###install Cloud9-hub to nginx config###"
 wget -O- https://raw.githubusercontent.com/HuJK/Cloud9Hub/master/c9io > /etc/nginx/sites-available/c9io
-ln -s ../sites-available/c9io /etc/nginx/sites-enabled/c9io
+ln -sfn ../sites-available/c9io /etc/nginx/sites-enabled/c9io
 cd /etc/c9
 
-echo "patch for login logout account and dashboard"
+echo "###patch for login logout account and dashboard###"
 wget -O- https://raw.githubusercontent.com/HuJK/Cloud9Hub/master/logout.patch | patch -p0
 wget -O- https://raw.githubusercontent.com/HuJK/Cloud9Hub/master/standalone.patch | patch -p0
-echo "patch for python3"
+echo "###patch for python3###"
 mkdir -p /etc/c9/.c9/runners/
 wget https://raw.githubusercontent.com/HuJK/Cloud9-hub/master/Python%203.run -O "/etc/c9/.c9/runners/Python 3.run"
 
-echo "generate self signed cert"
-apt-get install -y install openssl
+echo "###manually installing openssl v1.1.1a###"
```

Comment:
The better solution is use a valid certificate from let's encrypt or buy one. 
I use self signed cert here are just for expediency.
And the warning doesn't affect the generate of certs. Correct me if I am wrong.

I don't think it's worth to compile a new openssl just for generate a self-signed certificates without warning.

You should buy one or get one from let's encrypt.